port/common/omrsignal.c

```diff
@@ -166,11 +166,59 @@ omrsig_protect(struct OMRPortLibrary *portLibrary,  omrsig_protected_fn fn, void
  * @return 0 on success
  */
 uint32_t
-omrsig_set_async_signal_handler(struct OMRPortLibrary *portLibrary,  omrsig_handler_fn handler, void *handler_arg, uint32_t flags)
+omrsig_set_async_signal_handler(struct OMRPortLibrary *portLibrary, omrsig_handler_fn handler, void *handler_arg, uint32_t flags)
 {
 	return 1;
 }
 
+
+/*
+ * @brief Similar to omrsig_set_async_signal_handler. Refer to omrsig_set_async_signal_handler's description above.
+ * Differences: 1) one omrsig_handler_fn handler is registered with a signal at any time instead of multiple handlers.
+ * 2) The old OS handler is returned on success and NULL is returned on failure instead of returning 0 on success and
+ * non-zero on failure.
+ *
+ * @param[in] portLibrary The port library
+ * @param[in] handler the function to call if an asynchronous signal arrives
+ * @param[in] handler_arg the argument to handler
+ * @param[in] portlibSignalFlag port library signal flag
+ *
+ * @return old OS handler on success or NULL on failure
+ */
+void *
+omrsig_set_single_async_signal_handler(struct OMRPortLibrary *portLibrary, omrsig_handler_fn handler, void *handler_arg, uint32_t portlibSignalFlag)
+{
+	return NULL;
+}
+
+/*
+ * @brief Given an OS signal value, return the corresponding port library signal flag.
+ *
+ * @param[in] portLibrary The port library
+ * @param[in] osSignalValue OS signal value
+ *
+ * @return port library signal flag on success and 0 on failure
+*/
+uint32_t
+omrsig_map_os_signal_to_portlib_signal(struct OMRPortLibrary *portLibrary, uint32_t osSignalValue)
+{
+	return 0;
```

Comment:
Is `0` a signal value or are they all guaranteed to be `>` 0?

Reply:
If I execute `kill -l`, all signals are `>` than zero. I don't think I should use `0` for error since `SIG_ERR` is negative. I will change it to a negative number.

Reply:
`omrsig_map_os_signal_to_portlib_signal` returns a `port library signal flag`. a signal flag is used for bit operations so a signal flag will never be 0. so, 0 is acceptable as an error for `omrsig_map_os_signal_to_portlib_signal`.

`omrsig_map_portlib_signal_to_os_signal` returns an `OS signal value`. for this function, I changed the error to a negative value.